rust-runtime/aws-smithy-http-auth/src/auth.rs

```diff
@@ -187,11 +223,61 @@ impl HttpAuthDefinition {
     }
 }
 
+/// Types associated with [`HttpAuthDefinition`].
+pub mod http_auth_definition {
+    use super::AuthError;
+    use super::AuthErrorKind;
+    use super::HttpAuthDefinition;
+    use super::HttpAuthLocation;
+
+    /// A builder for [`HttpAuthDefinition`].
+    #[derive(Debug, Default)]
+    pub struct Builder {
+        location: HttpAuthLocation,
+        name: String,
+        scheme: Option<String>,
+    }
+
+    impl Builder {
+        /// Sets the HTTP auth location.
+        pub fn location(mut self, location: HttpAuthLocation) -> Self {
+            self.location = location;
+            self
+        }
+
+        /// Sets the the HTTP auth name.
+        pub fn name(mut self, name: impl Into<String>) -> Self {
+            self.name = name.into();
+            self
+        }
+
+        /// Sets the HTTP auth scheme.
+        pub fn scheme(mut self, scheme: impl Into<Option<String>>) -> Self {
```

Comment:
Our general pattern for this is for the main builder method to take a non-option, and then have a second `set_thing` method that takes option:
```rust
pub fn scheme(mut self, scheme: impl Into<String>) -> Self { ... }
pub fn set_scheme(&mut self, scheme: Option<String>) -> &mut Self { ... }
```

Also note the difference in the `self` types.

We follow this on non-optional fields too so that they can be unset. So the equivalent `set_` methods should be added to `location` and `name`.